### DISCUSSION TITLE
recognizing dependencies in buildSrc project with gradle.kts

### DISCUSSION THREAD

**Discussion by Monax111** [renovatebot/renovate#7953] [Request Help]
### Which Renovate are you using?

Renovate Open Source CLI

### Which platform are you using?

Bitbucket Server

### Have you checked the logs? Don't forget to include them if relevant
[fullLog.txt](https://github.com/renovatebot/config-help/files/4953332/fullLog.txt)

### What would you like to do?
I want update version spring in file buildSrc/build.gradle.kts
i find default regex for my template(gradle) '\\.gradle(\\.kts)?$'
i'ts look like it regex must find my file with twice match in it, but NO! i can't see any PR

my buildSrc/build.gradle.kts (trimmed)
```
plugins {
    id("org.springframework.boot") version "2.2.4.RELEASE" apply false
}

repositories {
    maven("https://registry.tcsbank.ru/repository/mvn-proxy-jcenter-release/")
}

kotlinDslPluginOptions {
    experimentalWarning.set(false)
}

dependencies {
    implementation("org.springframework.boot:spring-boot-gradle-plugin:2.2.4.RELEASE")
    
}
```
my config (trimmed)

```
"renovate-config": {
	"base-backend-kotlin": {
      "description": "Default base configuration for bpm kotlin backend applications",
      "extends": [
        ":ignoreUnstable",
        ":prImmediately",
        ":semanticCommits",
        ":semanticCommitTypeAll(renovatebot)",
        ":updateNotScheduled",
        ":automergeDisabled",
        ":ignoreModulesAndTests"
      ],
      "baseBranches": [
        "develop"
      ],
      "gradle": {
        "enable": true
      },
      "packageRules": [
        {
          "groupName": "jetbrains-kotlin",
          "packagePatterns": [
            "^org.jetbrains.kotlin"
          ]
        },
        {
          "groupName": "spring",
          "packagePatterns": [
            "^org.springframework:"
          ]
        },
        {
          "groupName": "spring-boot",
          "packagePatterns": [
            "^org.springframework.boot:"
          ]
        },
        {
          "groupName": "spring-cloud",
          "packagePatterns": [
            "^org.springframework.cloud:"
          ]
        },
       
      ]
    }
  }
```

**Comment by viceice** [comment]
Your log is truncated, we need the full log

**Comment by viceice** [comment]
Please provide a public reproduction repo

**Comment by Monax111** [comment]
@viceice, sorry, didn't notice what log is truncated
new log - [fullLog.txt](https://github.com/renovatebot/config-help/files/4959747/fullLog.txt)

About public reproduction repo: we have private bitbacked and private registry.
I create [repo ](https://github.com/Monax111/test), But we use custom preset (bpm-renovate:base-backend-kotlin) and i don't know how share it!. I have only source 
[config.txt](https://github.com/renovatebot/config-help/files/4959772/config.txt)

We start renovatebot with gitBush (file - start.sh)

is that enough or is it necessary to make a fully working project?

**Comment by Monax111** [comment]
in log i see my file:

DEBUG: Matched 15 file(s) for manager gradle: build.gradle.kts, **buildSrc/build.gradle.kts**, ...

but after i dont see what renovate try update version of plugin   id("org.springframework.boot") version "2.2.4.RELEASE" and version dependency "implementation("org.springframework.boot:spring-boot-gradle-plugin:2.2.4.RELEASE")"

**Comment by rarkins** [comment]
So is the title of this issue wrong?

**Comment by Monax111** [comment]
@rarkins  yes! But i dont know what is the problem to rename the title

**Comment by rarkins** [comment]
This is why a reproduction repository is important, otherwise we have little chance of guessing

**Comment by Monax111** [comment]
@rarkins  i do it, but it's update only version Gradle(
1) change token https://github.com/Monax111/test/blob/develop/config.js
2) run in console file start.sh

now i see error 
```
 "message": "Pushing to https://**redacted**@github.com': No such device or address\n",
         "stack": "Error: Pushing to https://**redacted**@github.com': No such device or address\n\n    at GitExecutorChain.onFatalException (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:59:77)\n    at GitExecutorChain.<anonymous> (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:51:21)\n    at Generator.throw (<anonymous>)\n    at rejected (/usr/src/app/node_modules/simple-git/src/lib/runners/git-executor-chain.js:6:65)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:97:5)"
```

**Comment by rarkins** [comment]
Sorry, but I don't understand

**Comment by Monax111** [comment]
@rarkins sorry, I had to explain in more detail
1) Now i have test project. Project on Spring+Gradle(kts)
2) I try use renovateBot for this project, and expect 5 PR:

build.gradle.kts: 
id("org.springframework.boot") version "2.2.8.RELEASE" -> "2.3.1.RELEASE"
id("io.spring.dependency-management") version "1.0.8.RELEASE" -> "1.0.9.RELEASE"
kotlin("jvm") version "1.3.60" ->"1.3.71"

 buildSrc/build.gradle.kts: 
id("org.springframework.boot") version "2.2.8.RELEASE"apply false -> "2.3.1.RELEASE"

gradle/wrapper/gradle-wrapper.properties
distributionUrl=https\://services.gradle.org/distributions/gradle-6.4.1-bin.zip -> 6.5.1

3) I use [self-host](https://docs.renovatebot.com/self-hosted-configuration/) to start renovateBot(see [file](https://github.com/Monax111/test/blob/develop/start.sh))
4) I have only one PR on gradle-wrapper. I would like to get 4 more PR after finish **start.sh**. In logs i see, whet renovate recognized old version, but can't push PR.
[fullLog.txt](https://github.com/renovatebot/config-help/files/4960870/fullLog.txt)

**Comment by rarkins** [comment]
Is that the same error that you get when you run on your private server?

**Comment by rarkins** [comment]
For example based on that error message, it's possible that the token you use does not have permissions to push

**Comment by Monax111** [comment]
> Is that the same error that you get when you run on your private server?

No! on the private server i get no errors( And no PR(((

Yes, it's my fault (extra space after the token)!.

Now i have 2 PR!! Progress))) 
I Expect PR on org.springframework.boot.gradle.plugin "2.2.8.RELEASE" -> "2.3.1.RELEASE" in file **build.gradle.kts** and **buildSrc/build.gradle.kts**. In log i see what renovate recognized it! 
```
 packageFiles with updates (repository=Monax111/test)
       "config": {
         "gradle": [
           {
             "packageFile": "build.gradle.kts",
             "datasource": "maven",
             "deps": [
               {
                 "name": "org.springframework.boot.gradle.plugin",
                 "depGroup": "org.springframework.boot",
                 "depName": "org.springframework.boot:org.springframework.boot.gradle.plugin",
                 "currentValue": "2.2.8.RELEASE",
                 "registryUrls": [
                   "https://repo.maven.apache.org/maven2/",
                   "https://plugins.gradle.org/m2"
                 ],
                 "depIndex": 0,
                 "updates": [
                   {
                     "fromVersion": "2.2.8.RELEASE",
                     "toVersion": "2.3.1.RELEASE",
                     "newValue": "2.3.1.RELEASE",
                     "newMajor": 2,
                     "newMinor": 3,
                     "updateType": "minor",
                     "isSingleVersion": true
                   }
                 ],
                 "warnings": [],
                 "sourceUrl": "https://github.com/spring-projects/spring-boot",
                 "homepage": "https://spring.io/projects/spring-boot"
               }, ...
```
But RP does not exist.

Same with kotlin("jvm") version "1.3.60" ->"1.3.71" - No PR, but logs contains update

[currentLog.txt](https://github.com/renovatebot/config-help/files/4961165/fullLog.txt)

**Comment by rarkins** [comment]
Looking at the log, it collapses 25 updates into 4 branches:

```
DEBUG: Returning 4 branch(es) (repository=Monax111/test)
```

There is an unrelated error you can ignore as it's caused by a changelog file being > 1MB:

```
DEBUG: Unknown GitHub error (repository=Monax111/test)
```

Further down you can see this:

```
DEBUG: Creation of new PRs is blocked by existing PRs (repository=Monax111/test)
       "prs": 0
```

So please run again after the next hour starts, or add `"prHourlyLimit": 0` to your `renovate.json`.

**Comment by Monax111** [comment]
`Looking at the log, it collapses 25 updates into 4 branches:`
Now i see 3 branch with  6 updates. updates = line?

`So please run again after the next hour starts, or add "prHourlyLimit": 0 to your renovate.json.`
  Yes! i add this parametr and now i have 4 change) But default value already 0?
https://docs.renovatebot.com/configuration-options/#prhourlylimit
looks very strange?

today i try to experiment with plugin:
1) now i have PR on update Spring(it's cool)
2) But i haven't PR on [second plugin](https://github.com/Monax111/test/blob/develop/buildSrc/build.gradle.kts#L2)
in [fullLog.txt](https://github.com/renovatebot/config-help/files/4965328/fullLog.txt) i can't find any about this plugin("com.gorylenko.gradle-git-properties") 
I think problen in gradle task "renovate" ? i can't find documentation about this task.

**Comment by rarkins** [comment]
> But default value already 0?

It is 2 in `config:base`, which you extend.

So the problem is that one dependency is never detected? Are any other dependencies detected in the same file?

**Comment by Monax111** [comment]
`So the problem is that one dependency is never detected? Are any other dependencies detected in the same file?`
Renovate recognized spring, and [create PR](https://github.com/Monax111/test/pull/15/files), but don't recognized git-properties plugin

**Comment by Monax111** [comment]
I think the problem may be that the gradle task Renovate does not collect dependencies from the buildSrc project. I apply Spring to the main project, but git-properties does not. Git-properties just define in buildSrc project.

**Comment by rarkins** [comment]
`config:base` ignores `test` directories by default. Please merge this so we can try without it: https://github.com/Monax111/test/pull/17

**Comment by Monax111** [comment]
@rarkins  
1) Merged
2) cleared old PR and branch
3) Start Renovate 
[fullLog.txt](https://github.com/renovatebot/config-help/files/4965498/fullLog.txt)
now i see 3 PR on Spring(PR are the same) and No Pr on git-properties plugin

**Comment by Monax111** [comment]
@rarkins i [move](https://github.com/Monax111/test/commit/6cfb9dfc47e1b62224cea555411ad797ae575b7c#diff-cc3ec92d4f7293c6ba674f5316e7a0c9) this plugin to root build and it work) i see PR on git-property! 
it's bug? Or this is the expected behavior?

**Comment by rarkins** [comment]
This seems like a bug to me. @ikesyo can you confirm?
